Mage/src/main/java/mage/target/common/TargetUpToCreaturePermanentAmount.java

```diff
@@ -0,0 +1,42 @@
+package mage.target.common;
+
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+
+/**
+ * @author Susucr
+ */
+public class TargetUpToCreaturePermanentAmount extends TargetPermanentAmount {
```

Comment:
if we're gonna have a special class for the "up to" case then it should probably be for any permanent rather than for just creatures